packages/@atjson/document/src/annotation.ts

```diff
@@ -49,6 +49,30 @@ export default abstract class Annotation<Attributes = {}> {
            this.end === annotation.end;
   }
 
+  compareAttributes(lhsAnnotationAttributes: any, rhsAnnotationAttributes: any) {
+    return Object.keys(lhsAnnotationAttributes).every(function (key) {
+      if (lhsAnnotationAttributes[key] !== rhsAnnotationAttributes[key]) {
+        if (lhsAnnotationAttributes[key] instanceof Document && rhsAnnotationAttributes[key] instanceof Document) {
+          return lhsAnnotationAttributes[key].equal(rhsAnnotationAttributes[key])
+        }
+        if (lhsAnnotationAttributes[key] typeof Object && rhsAnnotationAttributes[key] typeof Object) {
+          return this.compareAttributes(lhsAnnotationAttributes[key], rhsAnnotationAttributes[key])
+        }
+        return false;
+      }
+      return true;
+    });
+  }
+
+  equal(annotationToCompare: Annotation<any>) {
+    let lhsAnnotationAttributes = this.attributes;
+    let rhsAnnotationAttributes = annotationToCompare.attributes;
+    if (!(this.start === annotationToCompare.start && this.end === annotationToCompare.end && this.type === annotationToCompare.type)) {
```

Comment:
This could be turned into a series of `||` statements that may make this easier to grok.

```ts
return this.start === annotationToCompare.start &&
  this.end === annotationToCompare.end &&
  this.type === annotationToCompare.type &&
  areAttributesEqual(lhsAnnotationAttributes, rhsAnnotationAttributes);
```